src/app/static/src/app/components/WarningAlert.vue

```diff
@@ -0,0 +1,175 @@
+                default: 3,
+                required: false,
+                type: Number
+            }
+        },
+        data() {
+            return {
+                showFullBox: false,
+                fullBoxHeight: 0,
+                lineHeight: 0,
+                headerHeight: 0
+            };
+        },
+        computed: {
+            renderedBoxHeight(){
+                if (!this.warningsLengthy){
+                    return this.fullBoxHeight
+                } else {
+                    return this.showFullBox ? this.fullBoxHeight : this.maxBoxHeight - this.lineHeight
+                }
+            },
+            maxBoxHeight(){
+                return (this.maxLines * this.lineHeight) + this.headerHeight
+            },
+            warningsLengthy(){
+                return this.fullBoxHeight > this.maxBoxHeight
+            },
+            containerBox(){
+                return {
+                    height: `${this.renderedBoxHeight}px`,
+                    overflowY: "hidden"
+                }
+            },
+            filteredWarnings(){
+                const anObject: Warnings = {}
+                Object.keys(this.warnings).forEach((k) => {
+                    if (this.warnings[k].length > 0){
+                        anObject[k] = this.warnings[k]
+                    }
+                });
+                return anObject
+            },
+            currentLanguage: mapStateProp<RootState, Language>(
+                null,
+                (state: RootState) => state.language
+            ),
+            showAlert(){
+                return Object.keys(this.warnings).some(key => this.warnings[key].length > 0)
+            },
+            buttonText(){
+                return i18next.t(this.showFullBox ? "showLess" : "showMore", { lng: this.currentLanguage });
+            }
+        },
+        methods: {
+            toggleShowFullBox(){
+                this.showFullBox = !this.showFullBox;
+            },
+            updateDimensions(){
+                if (this.showAlert){
+                    this.lineHeight = (this.$refs.line as HTMLElement).clientHeight;
+                    this.headerHeight = (this.$refs.incHeader as HTMLElement).clientHeight - this.lineHeight;
+                    this.fullBoxHeight = (this.$refs.warningBox as HTMLElement).clientHeight;
+                }
+            },
+            headerText(key){
+                const headers: { [key: string]: string } = {
+                    modelOptions: "warningsHeaderModelOptions",
+                    modelRun: "warningsHeaderModelRun",
+                    modelCalibrate: "warningsHeaderModelCalibrate"
+                }
+                return key in headers ? i18next.t(headers[key], { lng: this.currentLanguage }) : ""
+            }
+        },
+        watch: {
+            warnings(){
+                this.updateDimensions()
+            }
+        },
+        mounted(){
+            this.updateDimensions()
+        },
```

Comment:
Because these updates only occur on mount or when the warnings prop changes, there is an edge case where the user could adjust their screen size after rendering and the height of the warning box will not be appropriate.

Reply:
They would have to adjust their screen size to the extent that the number of warning lines increases or decreases due to wrapping. So it's an unlikely scenario but I would be remiss if I didn't mention it.